### PR TITLE
fix: dev tools to 1.10 for Go 1.18

### PR DESCRIPTION
**Pull request by prabhu34**
New updates from renovate bots requires Go 1.18 which is not available in Developer Tools 1.7.
Also closes duplicate PR #105

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -18,7 +18,7 @@
 # Make will use bash instead of sh
 SHELL := /usr/bin/env bash
 
-DOCKER_TAG_VERSION_DEVELOPER_TOOLS := 1.7
+DOCKER_TAG_VERSION_DEVELOPER_TOOLS := 1.8
```
Comment (apeabody) [inline Makefile:21]:
```suggestion
DOCKER_TAG_VERSION_DEVELOPER_TOOLS := 1.10
```

File: build/lint.cloudbuild.yaml [hunk 2]
```diff
@@ -21,4 +21,4 @@ tags:
 - 'lint'
 substitutions:
   _DOCKER_IMAGE_DEVELOPER_TOOLS: 'cft/developer-tools'
-  _DOCKER_TAG_VERSION_DEVELOPER_TOOLS: '1.7'
+  _DOCKER_TAG_VERSION_DEVELOPER_TOOLS: '1.8'
```
Comment (apeabody) [inline build/lint.cloudbuild.yaml:24]:
```suggestion
  _DOCKER_TAG_VERSION_DEVELOPER_TOOLS: '1.10'
```

**Review by apeabody — COMMENTED** [review]
Thanks @prabhu34!

For GO `1.18` you will want dev tools `1.10`, however it's likely some other updates will be required for the newer linter.  If I recall correctly, dev tools `1.8` and `1.9` had GO `1.17`.

File: build/int.cloudbuild.yaml [hunk 3]
```diff
@@ -89,4 +89,4 @@ tags:
 - 'integration'
 substitutions:
   _DOCKER_IMAGE_DEVELOPER_TOOLS: 'cft/developer-tools'
-  _DOCKER_TAG_VERSION_DEVELOPER_TOOLS: '1.7'
+  _DOCKER_TAG_VERSION_DEVELOPER_TOOLS: '1.8'
```
[the remark this answers is not in the document]
Comment (prabhu34) [reply build/int.cloudbuild.yaml:92]: I have used 1.10 already in the latest commit

File: Makefile (hunk 1 above)
Comment (prabhu34) [reply Makefile:21]: I have used 1.10 already in the latest commit

File: build/lint.cloudbuild.yaml (hunk 2 above)
Comment (prabhu34) [reply build/lint.cloudbuild.yaml:24]: I have used 1.10 already in the latest commit

File: build/int.cloudbuild.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (prabhu34) [reply build/int.cloudbuild.yaml]: Oops! Yes I missed to push. Now I have done that.

**Comment by apeabody** [comment]
Thanks @prabhu34!  The new LINT in 1.10 is requiring a few additional changes if you can include:

The referenced but not used variables can normally be removed.  The lint can also be run locally with `make docker_test_lint`

```
Checking for tflint
Working in . ...
3 issue(s) found:

Warning: List items should be accessed using square brackets (terraform_deprecated_index)

  on main.tf line 108:
 108:       name        = var.generate_revision_name ? null : "${var.service_name}-${var.traffic_split.0.revision_name}"

Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_deprecated_index.md

Warning: List items should be accessed using square brackets (terraform_deprecated_index)

  on main.tf line 108:
 108:       name        = var.generate_revision_name ? null : "${var.service_name}-${var.traffic_split.0.revision_name}"

Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_deprecated_index.md

Warning: List items should be accessed using square brackets (terraform_deprecated_index)

  on main.tf line 108:
 108:       name        = var.generate_revision_name ? null : "${var.service_name}-${var.traffic_split.0.revision_name}"

Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_deprecated_index.md

tflint failed . 
Working in ./examples/cloud_run_vpc_connector ...
1 issue(s) found:

Warning: Interpolation-only expressions are deprecated in Terraform v0.12.14 (terraform_deprecated_interpolation)

  on network.tf line 45:
  45:     subnet_name     = "${module.vpc.subnets["us-central1/cloud-run-subnet"]["name"]}"

Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_deprecated_interpolation.md

tflint failed ./examples/cloud_run_vpc_connector 
Working in ./examples/secure_cloud_run ...
1 issue(s) found:

Warning: variable "ssl" is declared but not used (terraform_unused_declarations)

  on variables.tf line 17:
  17: variable "ssl" {

Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_unused_declarations.md

tflint failed ./examples/secure_cloud_run 
Working in ./examples/secure_cloud_run_standalone ...
2 issue(s) found:

Warning: variable "serverless_folder_suffix" is declared but not used (terraform_unused_declarations)

  on variables.tf line 33:
  33: variable "serverless_folder_suffix" {

Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_unused_declarations.md

Warning: variable "create_access_context_manager_access_policy" is declared but not used (terraform_unused_declarations)

  on variables.tf line 45:
  45: variable "create_access_context_manager_access_policy" {

Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_unused_declarations.md

tflint failed ./examples/secure_cloud_run_standalone 
Working in ./examples/simple_cloud_run ...
tflint passed ./examples/simple_cloud_run 
Working in ./examples/simple_cloud_run_with_cmek ...
tflint passed ./examples/simple_cloud_run_with_cmek 
Working in ./modules/secure-cloud-run ...
tflint passed ./modules/secure-cloud-run 
Working in ./modules/secure-cloud-run-core ...
2 issue(s) found:

Warning: Missing version constraint for provider "time" in "required_providers" (terraform_required_providers)

  on main.tf line 97:
  97: resource "time_sleep" "wait_30_seconds" {

Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_required_providers.md

Warning: `lb_name` variable has no type (terraform_typed_variables)

  on variables.tf line 130:
 130: variable "lb_name" {

Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_typed_variables.md

tflint failed ./modules/secure-cloud-run-core 
Working in ./modules/secure-cloud-run-harness ...
2 issue(s) found:

Warning: Missing version constraint for provider "random" in "required_providers" (terraform_required_providers)

  on service_perimeter.tf line 25:
  25: resource "random_id" "random_access_level_suffix" {

Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_required_providers.md

Warning: Missing version constraint for provider "time" in "required_providers" (terraform_required_providers)

  on service_perimeter.tf line 195:
 195: resource "time_sleep" "wait_90_seconds" {

Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_required_providers.md

tflint failed ./modules/secure-cloud-run-harness 
Working in ./modules/secure-cloud-run-net ...
1 issue(s) found:

Warning: module "firewall_rules" should specify a version (terraform_module_version)

  on firewall.tf line 22:
  22: module "firewall_rules" {

Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.1.1/docs/rules/terraform_module_version.md

tflint failed ./modules/secure-cloud-run-net 
```

**Comment by prabhu34** [comment]
@amandakarina Can you check the lint for the secure serverless modules? This is blocking the other PRs.

**Review by apeabody — APPROVED** [review]
Thanks @prabhu34 @amandakarina LGTM!

**Pull request merged**